core/src/main/java/org/apache/calcite/sql/type/OperandTypes.java

```diff
@@ -369,6 +369,49 @@ private boolean hasFractionalPart(BigDecimal bd) {
         }
       };
 
+  /**
+   * Operand type-checking strategy type must be a positive numeric non-NULL
+   * literal in the range 0 - 1.
+   */
+  public static final SqlSingleOperandTypeChecker POSITIVE_NUMERIC_LITERAL =
```

Comment:
Let's find a better name than 'positive'. Positive means 'greater than zero'. So your use of it here is misleading.

You seem to need something in the range 0 to 1, including both end points. That's a fairly common notion, so there must be a word for it.

Reply:
Maybe UNIT_INTERVAL_LITERAL suits here as the values are in [0,1].